ansible/playbooks/roles/containerd/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Remove Docker
+  block:
+  - name: Stop Kubelet before Docker removal
+    systemd:
+      name: kubelet
+      state: stopped
+    notify:
+    - Restart kubelet
+
+  - name: Remove Docker
+    include_role:
+      name: docker
+      tasks_from: remove-docker.yml
+  when:
+    - is_upgrade_run
+    - inventory_hostname not in groups.image_registry
+
+- name: Install Containerd package
+  package:
+    name: containerd.io
+    state: present
+  module_defaults:
+    yum: { lock_timeout: "{{ yum_lock_timeout }}" }
+
+- name: Configure prerequisites
+  include_tasks: configure-prerequisites.yml
+
+- name: Configure Containerd
+  include_tasks: configure-containerd.yml
+
+- name: Reconfigure kubelet args
+  when: is_upgrade_run
+  block:
+  - name: Get kubeadm-flags.env file content
+    slurp:
+      src: /var/lib/kubelet/kubeadm-flags.env
+    register: kubelet_kubeadm_args
+
+  - name: Set kubelet_kubeadmn_args_content
+    set_fact:
+      kubelet_kubeadmn_args_content: "{{ kubelet_kubeadm_args.content | b64decode }}"
+
+  - name: Modify container-runtime
```

Comment:
Please create an issue for refactoring this sequence of `replace` tasks, then add a comment here about it. I would see this e.g. as `from_env` parser (Ansible filter like we have `from_ini`) to load variables to Ansible dictionary, process it using Ansible, convert to env file format then replace the env file with one shot. Now we probably produce more backup files than needed (`replace` task in loop).